core/diagnostics/DiagnosticScenarios/Controllers/DiagnosticScenarios.cs

```diff
@@ -0,0 +1,162 @@
+using System;
+using System.Diagnostics;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using System.Threading;
+
+namespace testwebapi.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class DiagScenarioController : ControllerBase
+    {
+        object o1 = new object();
+        object o2 = new object();
+
+        private static Processor p = new Processor();
+
+        [HttpGet]
+        [Route("deadlock/")]
+        public ActionResult<string> deadlock()
+        {
+            (new System.Threading.Thread(() => {
+                DeadlockFunc();
+            })).Start();
+
+            Thread.Sleep(5000);
+
+            Thread[] threads = new Thread[300];
+            for(int i=0; i<300;i++)
+            {
+                (threads[i] = new Thread(() => {
+                    lock (o1) {Thread.Sleep(100);}
+                })).Start();
+            }
+
+            foreach(Thread thread in threads)
+            {
+                thread.Join();
+            }
+
+            return "success:deadlock";
+        }
+
+        private void DeadlockFunc()
+        {
+            lock (o1)
+            {
+                (new Thread(() => {
+                    lock (o2) { Monitor.Enter(o1); }
+                })).Start();
+
+                Thread.Sleep(2000);
+                Monitor.Enter(o2);
+            }
+        }
+
+        [HttpGet]
+        [Route("memspike/{seconds}")]
+        public ActionResult<string> memspike(int seconds)
+        {
+            Stopwatch watch=new Stopwatch();
```

Comment:
Line 63 should use `var`, In addition, It looks better to leave one space before and after the `=`.